src/python/pants/backend/python/goals/pytest_runner.py

```diff
@@ -151,9 +156,32 @@ async def run_all_setup_plugins(
 _EXTRA_OUTPUT_DIR = "extra-output"
 
 
+@dataclass(frozen=True)
+class TestMetadata:
+    """Parameters that must be constant for all test targets in a `pytest` batch."""
+
+    interpreter_constraints: InterpreterConstraints
+    extra_env_vars: tuple[str, ...]
+    xdist_concurrency: int | None
+    resolve: str
+    compatability_tag: str | None = None
```

Comment:
So, given that this must be constant, it feels like you could have an assertion somewhere that they are constant, but then just use the first FieldSet in a batch to get all the single value (since you've already asserted that they're equal)?

Reply:
We could - I was originally setting things up that way, but @thejcannon suggested I make use of the custom partitioning metadata type to store all the common bits within a batch. This was does save on (a little) duplicated work between the partitioner and runner rules, not sure how to weigh that against passing a fatter dataclass around.

Reply:
Very not a big deal.